pennylane/operation.py

```diff
@@ -798,6 +850,53 @@ class Observable(Operator):
     # pylint: disable=abstract-method
     return_type = None
 
+    @classmethod
+    def _eigvals(cls, *params):
+        """Eigenvalues of the observable.
+
+        The order of the eigenvalues needs to match the order of
```

Comment:
Did we check that this indeed holds for all relevant observables (I assume so)

Reply:
Otherwise all the integration tests would fail

Reply:
For `Identity`: cool, `np.array([1,1])`

The following use `pauli_eigs` (and have a set gateset for `diagonalizing_gates`):
* Hadamard
* PauliX
* PauliY
* PauliZ

For `Hermitian`:  `QubitUnitary` is being used with the eigenvectors and the eigenvalues are matched up with the correct eigenvectors (does not match, but that's no problem):
https://github.com/XanaduAI/pennylane/blob/f6689fe2ee115a7bfef954e7aea02b44b011445f/pennylane/ops/qubit.py#L1330

For `Tensor`: the eigenvalues are taken as the kronecker product for each wire and diagonalizing gates are merged together